hack/generator/pkg/codegen/pipeline_create_arm_types.go

```diff
@@ -123,17 +118,18 @@ func modifyKubeTypes(
 				return astmodel.TypeDefinition{}, errors.Errorf("couldn't find arm def matching kube def %q", def.Name())
 			}
 
-			return addArmConversionInterface(
-				def,
-				armDef,
-				idFactory,
-				false)
+			modifiedDef, err := addArmConversionInterface(def, armDef, idFactory, false)
+			if err != nil {
+				return astmodel.TypeDefinition{}, errors.Wrapf(err, "failed to add ARM conversion interface to %q", def.Name())
+			}
+
+			return *modifiedDef, nil
 		})
 }
 
 func iterDefs(
 	definitions astmodel.Types,
-	resourceHandler func(name astmodel.TypeName, resourceType *astmodel.ResourceType) (astmodel.TypeName, astmodel.TypeDefinition, error),
+	resourceHandler func(name astmodel.TypeName, resourceType *astmodel.ResourceType) (*astmodel.TypeName, *astmodel.TypeDefinition, error),
```

Comment:
Why change the name and definition returns to pointers here? It doesn't seem like you use it for anything (the place that calls `resourceHandler` just derefs them immediately), and the difference between the `resourceHandler` and `otherDefsHandler` is confusing. 

If it's just to simplify the error return lines, another way could be to add
```go
var noName astmodel.TypeName
var noDef astmodel.TypeDefinition
```
...at the top of the func and then use `return noName, noDef, errors.Wrapf(....)` on the error lines.

Reply:
Yeah, mostly a style issue - returning empty structs just feels as though the type system is forcing us to return _something_ when we really want to return _nothing_. 

At the risk of arguing by analogy, it's like going to the supermarket to buy a packet of chips and when your card is declined, you end up walking out with an empty packet instead of nothing at all.